### DISCUSSION TITLE
ArgoCD Helm Source - When is "helm template" performed?

### DISCUSSION THREAD

**Discussion by cristian98149** [argoproj/argo-cd#21197] [General]
I know that GitOps Engine performs a "helm template" in order to render manifest and apply them after the rendering.

The question is: When this happens? At every sync Argo runs an helm template in order to compare desired state and current on?

My deployments have "rollme: {{ randAlphaNum 5 | quote }} " annotation and I'm noticing that each 2 hours the value of this annotation changes, causing the POD to be recreated.

**Comment by pasha-codefresh** [comment]
It happens inside repo server, when controller doing reconciliation as part of it controller executes generatemanifests api call to repo server. During generatemanifests, repo server doing helm template

**Reply by pasha-codefresh** [reply]
https://github.com/argoproj/argo-cd/blob/433b317c358b1bfc59352791254d52e504d83dd4/reposerver/repository/repository.go#L1449

**Reply by crenshaw-dev** [reply]
> each 2 hours

That might be the "hard refresh" timeout.
